src/cmd_line/commands/file.ts

```diff
@@ -67,59 +78,76 @@ export class FileCommand extends node.CommandBase {
     };
 
     // No name was specified
-    if (this._arguments.name === undefined) {
-      if (this._arguments.createFileIfNotExists === true) {
+    if (this.arguments.name === undefined) {
+      if (this.arguments.createFileIfNotExists === true) {
         await vscode.commands.executeCommand('workbench.action.files.newUntitledFile');
         await hidePreviousEditor();
       }
       return;
     }
 
-    let filePath = '';
+    // Only untidify when the currently open page and file completion is local
+    if (this.arguments.name && editorFileUri.scheme === 'file') {
+      this._arguments.name = <string>untildify(this.arguments.name);
+    }
 
+    let fileUri = editorFileUri;
     // Using the empty string will request to open a file
-    if (this._arguments.name === '') {
+    if (this.arguments.name === '') {
       // No name on split is fine and just return
       if (split === true) {
         return;
       }
 
       const fileList = await vscode.window.showOpenDialog({});
-      if (fileList) {
-        filePath = fileList[0].fsPath;
+      if (fileList && fileList.length > 0) {
+        fileUri = fileList[0];
       }
     } else {
-      // Using a filename, open or create the file
-      this._arguments.name = this._arguments.name.replace(/^file:\/\//, '');
+      // remove file://
+      this._arguments.name = this.arguments.name.replace(/^file:\/\//, '');
 
-      filePath = path.isAbsolute(this._arguments.name)
-        ? path.normalize(this._arguments.name)
-        : path.join(path.dirname(editorFilePath), this._arguments.name);
+      // Using a filename, open or create the file
+      const { fullPath, path: p } = getFullPath(this.arguments.name);
+      // Only if the expanded path of the full path is different than
+      // the currently opened window path
+      if (fullPath !== editorFilePath) {
+        const uriPath = getUriPath(editorFileUri, fullPath, p.sep);
+        if (uriPath === null) {
+          // return the if path is invalid
+          return;
+        }
 
-      if (filePath !== editorFilePath) {
-        let fileExists = await doesFileExist(filePath);
+        let fileExists = await doesFileExist(uriPath);
         if (!fileExists) {
           // if file does not exist
           // try to find it with the same extension as the current file
-          const pathWithExt = filePath + path.extname(editorFilePath);
-          fileExists = await doesFileExist(pathWithExt);
-          if (fileExists) {
-            filePath = pathWithExt;
+          const pathWithExt = fullPath + p.extname(editorFilePath);
+          const uriPathWithExt = getUriPath(editorFileUri, pathWithExt, p.sep);
+          if (uriPathWithExt !== null) {
+            fileExists = await doesFileExist(uriPathWithExt);
+            if (fileExists) {
+              // if the file with the added ext exists
+              fileUri = uriPathWithExt;
+            }
           }
+        } else {
```

Comment:
can you reverse the logic?

if (fileExists) {
} else {
}

always easier to read affirmation over negation.

Reply:
Agree changed